src/layered_config_tree/main.py

```diff
@@ -229,10 +237,16 @@ class LayeredConfigTree:
 
     """
 
+    # Define type annotations here since they're indirectly defined below
+    _layers: list[str]
+    _children: dict[str, Union["LayeredConfigTree", "ConfigNode"]]
+    _frozen: bool
+    _name: str
+
     def __init__(
         self,
-        data: Union[Dict, str, Path, "LayeredConfigTree"] = None,
-        layers: List[str] = None,
+        data: Optional[Union[NestedDict, str, Path, "LayeredConfigTree"]] = None,
```

Comment:
This type `Union[NestedDict, str, Path, "LayeredConfigTree"]` is reused in a lot of places. It might be useful to define this with a type alias?